pkg/autoupgrade/daemon_test.go

```diff
@@ -25,6 +25,7 @@ type mockDaemonClient struct {
 	localTags, remoteTags               []string
 	remoteImageDigest, resolvedLocalTag string
 	localTagFound                       bool
+	imageDenyList                       map[string]any
```

Comment:
The way I usually do this in Go is 
```suggestion
	imageDenyList                       map[string]struct{}
```
Then, when you check for existence below...